sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -543,6 +716,80 @@ graph LR
     Modifiable -->|Finalize| Executable
 ....
 
+==== Executable Graph Update
+
+A graph in the executable state can have the configuration of its nodes modified
+using a concept called graph _update_. This avoids a user having to rebuild and
+finalize a new executable graph when only the inputs & outputs to a graph
+change between submissions. Updates to a node take effect from the next
+submission of a graph and will not affect any previous submissions or in-flight
+executions of the same graph.
+
+The only type of node that is currently supported for updating in a graph is
+kernel execution nodes.
+
+The aspects of a kernel execution node that can be configured during update are:
+
+* Parameters to the kernel.
+* ND-Range of the kernel.
+
+Two methods are provided by the API to the user for performing this update.
+An API for updating the complete graph object, which is most useful when the
+graph was created from a library, and an individual node update API.
+
+To update an executable graph, the `property::graph::updatable_graph` property
+must have been set when the graph was created during finalization. Otherwise, an
+exception will be thrown if a user tries to update an executable graph. This
+guarantee allows the backend to provide a more optimized implementation, if
+possible.
+
+The `property::graph::updatable_graph` property also allows an executable graph
+to be submitted for execution while a previous submission of the same
+executable graph instance is still executing. This is because the ability to
+change the graph inputs/outputs can remove the data race conditions that could
+otherwise exist if the same executable graph was executed concurrently.
+
+===== Whole Graph Update
+
+===== Individual Node Update
+
+Memory parameters to individual nodes in a graph in the `executable` state
+can be updated between graph executions using dynamic parameters. When a
+`dynamic_parameter` instance is created it is associated with a memory object.
+By registering the `dynamic_parameter` with a node using
+`node::register_dynamic_parameter()` the memory object underlying dynamic parameter
+is matched to a node argument, and the `dynamic_parameter` can then be
+used to update that node argument in future. This registration is necessary
+because SYCL kernel arguments don't have indices and the argument ordering of
+lambda captures isn't defined, so we need a way to tie the underlying node
+argument to a `dynamic_parameter` object.
+
+The update itself is then performed on the `dynamic_parameter` by calling
+`command_graph<graph_state::executable>::update_dynamic_parameter()`. If
+a user tries to register a dynamic parameter with a node, after that dynamic
```

Comment:
does this sentence mean that: once command_graph<graph_state::executable>::update_dynamic_parameter(dynamic_parameter dynamicParam, T newValue) is returned, the dynamicParam will contain the newValue?